src/pybind/mgr/dashboard/module.py

```diff
@@ -46,6 +46,80 @@ def fixed_init(hc_self, server, sock, makefile=CP_fileobject):
 
         HTTPConnection.__init__ = fixed_init
 
+    # cherrypy.wsgiserver was extracted wsgiserver into cheroot in cherrypy v9.0.0
+    def patch_builtin_ssl_wrap(new_wrap):
+        if v < StrictVersion("9.0.0"):
+            from cherrypy.wsgiserver.ssl_builtin import BuiltinSSLAdapter as builtin_ssl
+        else:
+            from cherrypy.cheroot.ssl.builtin import BuiltinSSLAdapter as builtin_ssl
+        builtin_ssl.wrap = new_wrap(builtin_ssl.wrap)
+
+    # the fix was included by cheroot v5.2.0, which was included by cherrypy
+    # 10.2.0.
+    if v < StrictVersion("10.2.0"):
+        # see https://github.com/cherrypy/cheroot/pull/4
+        import ssl
+
+        def accept_ssl_errors(func):
+            def wrapper(self, sock):
+                try:
+                    return func(self, sock)
+                except ssl.SSLError as e:
+                    if e.errno == ssl.SSL_ERROR_SSL:
+                        # Check if it's one of the known errors
+                        # Errors that are caught by PyOpenSSL, but thrown by
+                        # built-in ssl
+                        _block_errors = ('unknown protocol', 'unknown ca',
+                                         'unknown_ca', 'inappropriate fallback',
+                                         'wrong version number',
+                                         'no shared cipher', 'certificate unknown',
+                                         'ccs received early')
+                        for error_text in _block_errors:
+                            if error_text in e.args[1].lower():
+                                # Accepted error, let's pass
+                                return None, {}
+                        raise
+            return wrapper
+        patch_builtin_ssl_wrap(accept_ssl_errors)
+
+    try:
+        import cheroot
+        cheroot_version = cheroot.__version__
+        del cheroot
```

Comment:
What do you want to achieve here? Looks like you want to unload a module by just deleting it. That doesn't work

```
In [5]: 'requests' in sys.modules
Out[5]: False

In [6]: import requests

In [7]: 'requests' in sys.modules
Out[7]: True

In [8]: del requests

In [9]: 'requests' in sys.modules
Out[9]: True
```

Reply:
@sebastian-philipp i don't think python suppport unload a loaded module. what i want to do is to port the fix from upstream and follow their convention. the practical reason is, to avoid polluting the naming space.